### PR TITLE
[MIG] quality_control_stock: Migration to 12.0

### PR DESCRIPTION
**Pull request by NachoAlesLopez**
cc @guadaltech

### CODE REVIEW

File: quality_control_stock/models/qc_inspection.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        res = super(QcInspection, self)._prepare_inspection_header(
+            object_ref, trigger_line)
+        # Fill qty when coming from pack operations
+        if object_ref and object_ref._name == 'stock.move.line':
+            res['qty'] = object_ref.product_qty
+        if object_ref and object_ref._name == 'stock.move':
+            res['qty'] = object_ref.product_uom_qty
+        return res
+
+    picking_id = fields.Many2one(
```
Comment (rousseldenis) [inline quality_control_stock/models/qc_inspection.py]: Please put fields definition on top

File: quality_control_stock/models/qc_inspection.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    inspection.picking_id = inspection.object_id.picking_id
+                elif inspection.object_id._name == 'stock.picking':
+                    inspection.picking_id = inspection.object_id
+                elif inspection.object_id._name == 'stock.move.line':
+                    inspection.picking_id = inspection.object_id.picking_id
+
+    @api.multi
+    @api.depends('object_id')
+    def _compute_lot(self):
+        for inspection in self:
```
Comment (rousseldenis) [inline quality_control_stock/models/qc_inspection.py:47]: Please put the search before loop:

```suggestion
        moves = self.filtered(lambda i: i.object_id._name == 'stock.move')
        move_lines = self.env['stock.move.line'].search([
             ('lot_id', '!=', False),
             ('move_id', 'in', moves.ids)])
```

Then in 'elif inspection.object_id._name == 'stock.move':' :
```suggestion
        from odoo.fields import first  # (on top)
        inspection.lot_id = first(move_lines.filtered(lambda l: l.move_id == inspection.object_id))
```

This avoid to do too much queries.

File: quality_control_stock/models/qc_inspection.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    def object_selection_values(self):
+        result = super(QcInspection, self).object_selection_values()
+        result.extend([('stock.picking', "Picking List"), ('stock.move', "Stock Move")])
+        return result
+
+    @api.multi
+    @api.depends('object_id')
+    def _compute_picking(self):
+        for inspection in self:
+            inspection.picking_id = False
```
Comment (rousseldenis) [inline quality_control_stock/models/qc_inspection.py]: Computed values are False by default before compute call. So, if value is not set, False will be.

File: quality_control_stock/models/qc_inspection.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+from odoo import api, fields, models
+
+
+class QcInspection(models.Model):
+    _inherit = 'qc.inspection'
+
+    @api.multi
+    def object_selection_values(self):
+        result = super(QcInspection, self).object_selection_values()
```
Comment (rousseldenis) [inline quality_control_stock/models/qc_inspection.py]:
```suggestion
        result = super().object_selection_values()
```

File: quality_control_stock/models/qc_trigger.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class QcTrigger(models.Model):
+    _inherit = 'qc.trigger'
+
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", readonly=True, ondelete="cascade")
```
Comment (rousseldenis) [inline quality_control_stock/models/qc_trigger.py:12]: readonly is True by default in v12

File: quality_control_stock/models/stock_picking.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        compute="_compute_count_inspections", string="Inspections OK")
+    failed_inspections = fields.Integer(
+        compute="_compute_count_inspections", string="Inspections failed")
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        inspection_model = self.env['qc.inspection']
+        for operation in self.move_lines:
+            qc_trigger = self.env['qc.trigger'].search(
```
Comment (rousseldenis) [inline quality_control_stock/models/stock_picking.py]: You must call this search before the loop

File: quality_control_stock/models/stock_picking_type.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        picking_type._create_qc_trigger()
+        return picking_type
+
+    @api.multi
+    def write(self, vals):
+        res = super(StockPickingType, self).write(vals)
+        if vals.get('name') or vals.get('warehouse_id'):
+            qc_trigger_model = self.env['qc.trigger'].sudo()
+            for rec in self:
+                qc_triggers = qc_trigger_model.search(
```
Comment (rousseldenis) [inline quality_control_stock/models/stock_picking_type.py:34]: You should call the search before the loop and then filter by picking_type.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some optimization changes.

**Comment by NachoAlesLopez** [comment]
@rousseldenis thanks for the suggestions! I've applied the improvements.

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍 

Seems to be OK

**Comment by rousseldenis** [comment]
@NachoAlesLopez A little lint error and ok

**Comment by NachoAlesLopez** [comment]
> @NachoAlesLopez A little lint error and ok

Fixed.

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:

**Comment by MiquelRForgeFlow** [comment]
conflicts

**Comment by HviorForgeFlow** [comment]
> conflicts

up!

**Comment by LoisRForgeFlow** [comment]
I just rebased the branch to solve conflicts.

**Comment by LoisRForgeFlow** [comment]
@rousseldenis could you update your review?

**Review by HviorForgeFlow — APPROVED** [review]

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
I got this error when updating point_of_sale module on a 12 db database with this module:

> 2019-07-26 08:56:51,333 25096 INFO test_db odoo.modules.registry: module point_of_sale: creating or updating database tables 
> 2019-07-26 08:56:52,149 25096 INFO test_db odoo.modules.loading: loading point_of_sale/security/point_of_sale_security.xml 
> 2019-07-26 08:56:52,267 25096 INFO test_db odoo.modules.loading: loading point_of_sale/security/ir.model.access.csv 
> 2019-07-26 08:56:52,478 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/default_barcode_patterns.xml 
> 2019-07-26 08:56:52,497 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/digest_data.xml 
> 2019-07-26 08:56:52,502 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_box.xml 
> 2019-07-26 08:56:52,513 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_details.xml 
> 2019-07-26 08:56:52,540 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_open_statement.xml 
> 2019-07-26 08:56:52,572 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_payment.xml 
> 2019-07-26 08:56:52,599 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_templates.xml 
> 2019-07-26 08:56:53,144 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_template.xml 
> 2019-07-26 08:56:53,491 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_report.xml 
> 2019-07-26 08:56:53,537 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_view.xml 
> 2019-07-26 08:56:53,591 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_order_view.xml 
> 2019-07-26 08:56:53,837 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_category_view.xml 
> 2019-07-26 08:56:53,905 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/product_view.xml 
> 2019-07-26 08:56:54,218 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/account_journal_view.xml 
> 2019-07-26 08:56:54,289 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_config_view.xml 
> 2019-07-26 08:56:54,539 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_session_view.xml 
> 2019-07-26 08:56:54,680 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_sequence.xml 
> 2019-07-26 08:56:54,685 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale.xml 
> 2019-07-26 08:56:54,754 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/point_of_sale_data.xml 
> 2019-07-26 08:56:54,777 25096 WARNING test_db odoo.modules.loading: Transient module states were reset 
> 2019-07-26 08:56:54,778 25096 ERROR test_db odoo.modules.registry: Failed to load registry 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/models.py", line 4718, in ensure_one
>     _id, = self._ids
> ValueError: not enough values to unpack (expected 1, got 0)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
>     self._tags[rec.tag](rec, de, mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
>     record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
>   File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
>     records = self._load_records_create([data['values'] for data in to_create])
>   File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
>     return self.create(values)
>   File "<decorator-gen-284>", line 2, in create
>   File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
>     return create(self, arg)
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
>     picking_types._create_qc_trigger()
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
>     self.ensure_one()
>   File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
>     raise ValueError("Expected singleton: %s" % self)
> ValueError: Expected singleton: stock.picking.type()
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/modules/registry.py", line 86, in new
>     odoo.modules.load_modules(registry._db, force_demo, status, update_module)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 417, in load_modules
>     force, status, report, loaded_modules, update_module, models_to_check)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 313, in load_marked_modules
>     perform_checks=perform_checks, models_to_check=models_to_check
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 222, in load_module_graph
>     load_data(cr, idref, mode, kind='data', package=package, report=report)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 68, in load_data
>     tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 802, in convert_file
>     convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 865, in convert_xml_import
>     obj.parse(doc.getroot(), mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 755, in parse
>     self.parse(rec, mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 764, in parse
>     exc_info[2]
>   File "/opt/odoo/OCB12/odoo/tools/pycompat.py", line 86, in reraise
>     raise value.with_traceback(tb)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
>     self._tags[rec.tag](rec, de, mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
>     record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
>   File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
>     records = self._load_records_create([data['values'] for data in to_create])
>   File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
>     return self.create(values)
>   File "<decorator-gen-284>", line 2, in create
>   File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
>     return create(self, arg)
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
>     picking_types._create_qc_trigger()
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
>     self.ensure_one()
>   File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
>     raise ValueError("Expected singleton: %s" % self)
> odoo.tools.convert.ParseError: "Expected singleton: stock.picking.type()" while parsing /opt/odoo/OCB12/addons/point_of_sale/data/point_of_sale_data.xml:12, near
> <record id="picking_type_posout" model="stock.picking.type">
>             <field name="name">PoS Orders</field>
>             <field name="sequence_id" ref="seq_picking_type_posout"/>
>             <field name="default_location_src_id" ref="stock.stock_location_stock"/>
>             <field name="default_location_dest_id" ref="stock.stock_location_customers"/>
>             <field name="warehouse_id" eval="False"/>
>             <field name="code">outgoing</field>
>         </record>
> 2019-07-26 08:56:54,783 25096 CRITICAL test_db odoo.service.server: Failed to initialize database `test_db`. 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/models.py", line 4718, in ensure_one
>     _id, = self._ids
> ValueError: not enough values to unpack (expected 1, got 0)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
>     self._tags[rec.tag](rec, de, mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
>     record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
>   File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
>     records = self._load_records_create([data['values'] for data in to_create])
>   File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
>     return self.create(values)
>   File "<decorator-gen-284>", line 2, in create
>   File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
>     return create(self, arg)
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
>     picking_types._create_qc_trigger()
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
>     self.ensure_one()
>   File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
>     raise ValueError("Expected singleton: %s" % self)
> ValueError: Expected singleton: stock.picking.type()
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/opt/odoo/OCB12/odoo/service/server.py", line 1116, in preload_registries
>     registry = Registry.new(dbname, update_module=update_module)
>   File "/opt/odoo/OCB12/odoo/modules/registry.py", line 86, in new
>     odoo.modules.load_modules(registry._db, force_demo, status, update_module)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 417, in load_modules
>     force, status, report, loaded_modules, update_module, models_to_check)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 313, in load_marked_modules
>     perform_checks=perform_checks, models_to_check=models_to_check
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 222, in load_module_graph
>     load_data(cr, idref, mode, kind='data', package=package, report=report)
>   File "/opt/odoo/OCB12/odoo/modules/loading.py", line 68, in load_data
>     tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 802, in convert_file
>     convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 865, in convert_xml_import
>     obj.parse(doc.getroot(), mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 755, in parse
>     self.parse(rec, mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 764, in parse
>     exc_info[2]
>   File "/opt/odoo/OCB12/odoo/tools/pycompat.py", line 86, in reraise
>     raise value.with_traceback(tb)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
>     self._tags[rec.tag](rec, de, mode=mode)
>   File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
>     record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
>   File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
>     records = self._load_records_create([data['values'] for data in to_create])
>   File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
>     return self.create(values)
>   File "<decorator-gen-284>", line 2, in create
>   File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
>     return create(self, arg)
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
>     picking_types._create_qc_trigger()
>   File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
>     self.ensure_one()
>   File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
>     raise ValueError("Expected singleton: %s" % self)
> odoo.tools.convert.ParseError: "Expected singleton: stock.picking.type()" while parsing /opt/odoo/OCB12/addons/point_of_sale/data/point_of_sale_data.xml:12, near
> <record id="picking_type_posout" model="stock.picking.type">
>             <field name="name">PoS Orders</field>
>             <field name="sequence_id" ref="seq_picking_type_posout"/>
>             <field name="default_location_src_id" ref="stock.stock_location_stock"/>
>             <field name="default_location_dest_id" ref="stock.stock_location_customers"/>
>             <field name="warehouse_id" eval="False"/>
>             <field name="code">outgoing</field>
>         </record>
> 2019-07-26 08:56:54,785 25096 INFO test_db odoo.service.server: Initiating shutdown 
> 2019-07-26 08:56:54,785 25096 INFO test_db odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown. 
> 
> Process finished with exit code 255

**Comment by NachoAlesLopez** [comment]
> I got this error when updating point_of_sale module on a 12 db database with this module:
> 
> > 2019-07-26 08:56:51,333 25096 INFO test_db odoo.modules.registry: module point_of_sale: creating or updating database tables
> > 2019-07-26 08:56:52,149 25096 INFO test_db odoo.modules.loading: loading point_of_sale/security/point_of_sale_security.xml
> > 2019-07-26 08:56:52,267 25096 INFO test_db odoo.modules.loading: loading point_of_sale/security/ir.model.access.csv
> > 2019-07-26 08:56:52,478 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/default_barcode_patterns.xml
> > 2019-07-26 08:56:52,497 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/digest_data.xml
> > 2019-07-26 08:56:52,502 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_box.xml
> > 2019-07-26 08:56:52,513 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_details.xml
> > 2019-07-26 08:56:52,540 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_open_statement.xml
> > 2019-07-26 08:56:52,572 25096 INFO test_db odoo.modules.loading: loading point_of_sale/wizard/pos_payment.xml
> > 2019-07-26 08:56:52,599 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_templates.xml
> > 2019-07-26 08:56:53,144 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_template.xml
> > 2019-07-26 08:56:53,491 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_report.xml
> > 2019-07-26 08:56:53,537 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_view.xml
> > 2019-07-26 08:56:53,591 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_order_view.xml
> > 2019-07-26 08:56:53,837 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_category_view.xml
> > 2019-07-26 08:56:53,905 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/product_view.xml
> > 2019-07-26 08:56:54,218 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/account_journal_view.xml
> > 2019-07-26 08:56:54,289 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_config_view.xml
> > 2019-07-26 08:56:54,539 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/pos_session_view.xml
> > 2019-07-26 08:56:54,680 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale_sequence.xml
> > 2019-07-26 08:56:54,685 25096 INFO test_db odoo.modules.loading: loading point_of_sale/views/point_of_sale.xml
> > 2019-07-26 08:56:54,754 25096 INFO test_db odoo.modules.loading: loading point_of_sale/data/point_of_sale_data.xml
> > 2019-07-26 08:56:54,777 25096 WARNING test_db odoo.modules.loading: Transient module states were reset
> > 2019-07-26 08:56:54,778 25096 ERROR test_db odoo.modules.registry: Failed to load registry
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/models.py", line 4718, in ensure_one
> > _id, = self._ids
> > ValueError: not enough values to unpack (expected 1, got 0)
> > During handling of the above exception, another exception occurred:
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
> > self._tags[rec.tag](rec, de, mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
> > record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
> > File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
> > records = self._load_records_create([data['values'] for data in to_create])
> > File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
> > return self.create(values)
> > File "", line 2, in create
> > File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
> > return create(self, arg)
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
> > picking_types._create_qc_trigger()
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
> > self.ensure_one()
> > File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
> > raise ValueError("Expected singleton: %s" % self)
> > ValueError: Expected singleton: stock.picking.type()
> > During handling of the above exception, another exception occurred:
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/modules/registry.py", line 86, in new
> > odoo.modules.load_modules(registry._db, force_demo, status, update_module)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 417, in load_modules
> > force, status, report, loaded_modules, update_module, models_to_check)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 313, in load_marked_modules
> > perform_checks=perform_checks, models_to_check=models_to_check
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 222, in load_module_graph
> > load_data(cr, idref, mode, kind='data', package=package, report=report)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 68, in load_data
> > tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 802, in convert_file
> > convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 865, in convert_xml_import
> > obj.parse(doc.getroot(), mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 755, in parse
> > self.parse(rec, mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 764, in parse
> > exc_info[2]
> > File "/opt/odoo/OCB12/odoo/tools/pycompat.py", line 86, in reraise
> > raise value.with_traceback(tb)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
> > self._tags[rec.tag](rec, de, mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
> > record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
> > File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
> > records = self._load_records_create([data['values'] for data in to_create])
> > File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
> > return self.create(values)
> > File "", line 2, in create
> > File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
> > return create(self, arg)
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
> > picking_types._create_qc_trigger()
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
> > self.ensure_one()
> > File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
> > raise ValueError("Expected singleton: %s" % self)
> > odoo.tools.convert.ParseError: "Expected singleton: stock.picking.type()" while parsing /opt/odoo/OCB12/addons/point_of_sale/data/point_of_sale_data.xml:12, near
> > 
> > PoS Orders
> > 
> > 
> > 
> > 
> > outgoing
> > 
> > 2019-07-26 08:56:54,783 25096 CRITICAL test_db odoo.service.server: Failed to initialize database `test_db`.
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/models.py", line 4718, in ensure_one
> > _id, = self._ids
> > ValueError: not enough values to unpack (expected 1, got 0)
> > During handling of the above exception, another exception occurred:
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
> > self._tags[rec.tag](rec, de, mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
> > record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
> > File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
> > records = self._load_records_create([data['values'] for data in to_create])
> > File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
> > return self.create(values)
> > File "", line 2, in create
> > File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
> > return create(self, arg)
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
> > picking_types._create_qc_trigger()
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
> > self.ensure_one()
> > File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
> > raise ValueError("Expected singleton: %s" % self)
> > ValueError: Expected singleton: stock.picking.type()
> > During handling of the above exception, another exception occurred:
> > Traceback (most recent call last):
> > File "/opt/odoo/OCB12/odoo/service/server.py", line 1116, in preload_registries
> > registry = Registry.new(dbname, update_module=update_module)
> > File "/opt/odoo/OCB12/odoo/modules/registry.py", line 86, in new
> > odoo.modules.load_modules(registry._db, force_demo, status, update_module)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 417, in load_modules
> > force, status, report, loaded_modules, update_module, models_to_check)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 313, in load_marked_modules
> > perform_checks=perform_checks, models_to_check=models_to_check
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 222, in load_module_graph
> > load_data(cr, idref, mode, kind='data', package=package, report=report)
> > File "/opt/odoo/OCB12/odoo/modules/loading.py", line 68, in load_data
> > tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 802, in convert_file
> > convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 865, in convert_xml_import
> > obj.parse(doc.getroot(), mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 755, in parse
> > self.parse(rec, mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 764, in parse
> > exc_info[2]
> > File "/opt/odoo/OCB12/odoo/tools/pycompat.py", line 86, in reraise
> > raise value.with_traceback(tb)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 758, in parse
> > self._tags[rec.tag](rec, de, mode=mode)
> > File "/opt/odoo/OCB12/odoo/tools/convert.py", line 663, in _tag_record
> > record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
> > File "/opt/odoo/OCB12/odoo/models.py", line 3865, in _load_records
> > records = self._load_records_create([data['values'] for data in to_create])
> > File "/opt/odoo/OCB12/odoo/models.py", line 3779, in _load_records_create
> > return self.create(values)
> > File "", line 2, in create
> > File "/opt/odoo/OCB12/odoo/api.py", line 452, in _model_create_multi
> > return create(self, arg)
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 25, in create
> > picking_types._create_qc_trigger()
> > File "/opt/odoo/manufacture12/quality_control_stock/models/stock_picking_type.py", line 13, in _create_qc_trigger
> > self.ensure_one()
> > File "/opt/odoo/OCB12/odoo/models.py", line 4721, in ensure_one
> > raise ValueError("Expected singleton: %s" % self)
> > odoo.tools.convert.ParseError: "Expected singleton: stock.picking.type()" while parsing /opt/odoo/OCB12/addons/point_of_sale/data/point_of_sale_data.xml:12, near
> > 
> > PoS Orders
> > 
> > 
> > 
> > 
> > outgoing
> > 
> > 2019-07-26 08:56:54,785 25096 INFO test_db odoo.service.server: Initiating shutdown
> > 2019-07-26 08:56:54,785 25096 INFO test_db odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.
> > Process finished with exit code 255

@hveficent This error should be fixed now. Can you check it out?
@lreficent Thanks for taking care of the merge conflict!

**Comment by pedrobaeza** [comment]
Please include #393

**Comment by NachoAlesLopez** [comment]
> Please include #393

Done.

**Pull request merged**